### PR TITLE
web replay

### PR DESCRIPTION
**Pull request by jasonLaster**

### CODE REVIEW

File: _posts/2018-09-18-webreplay.markdown [hunk 1]
```diff
@@ -0,0 +1,99 @@
+### Record and Replay: An overview for web developers like me
+
+Brian Hackett recently published this brilliant [patch][patch], which prompted me to think a bit more about Web Replay’s architecture. Yesterday, I had the opportunity to talk with him. Here are some of my notes. If you're interested in learning more, the [MDN] has an in-depth overview.
+
+
+### High Level
+
+Web Replay records Firefox’s system calls while the browser is running. Later, when replaying it reads from the recording.
```
Comment (darkwing) [inline _posts/2018-09-18-webreplay.markdown:8]: "it" is ambiguous here; I'd use a more descriptive word.

File: _posts/2018-09-18-webreplay.markdown [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  MACRO(read, SaveRvalHadErrorNegative<WriteBufferViaRval<1, 2>>) \
+  MACRO(open, SaveRvalHadErrorNegative)                       \
+```
+
+
+### Intercepting Calls
+
+Web Replay does not instrument the browser, it overwrites system calls. Put another way, it doesn't observe the OS, it monkey patches it. What could go wrong, right?!?!
+
+Web Replay maps the system call locations to a single Web Replay assembly function appropariately called `RecordReplayRedirectCall` in [ProcessRedirect.cpp][ProcessRedirect.cpp]. If you were wondering, "when are we going to get to the inlined assembly code?", now is the time! All kidding aside, if you're like me and have never seen assembly in the wild, here is a [video][video] series on assembly that I started watching this weekend, which is pretty great.
```
Comment (darkwing) [inline _posts/2018-09-18-webreplay.markdown]: appropariately - spelling

File: _posts/2018-09-18-webreplay.markdown [hunk 3]
```diff
@@ -0,0 +1,99 @@
+### Record and Replay: An overview for web developers like me
+
+Brian Hackett recently published this brilliant [patch][patch], which prompted me to think a bit more about Web Replay’s architecture. Yesterday, I had the opportunity to talk with him. Here are some of my notes. If you're interested in learning more, the [MDN] has an in-depth overview.
```
Comment (juliandescottes) [inline _posts/2018-09-18-webreplay.markdown]: `the [MDN] has` -> `[MDN] has`

File: _posts/2018-09-18-webreplay.markdown [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+
+### High Level
+
+Web Replay records Firefox’s system calls while the browser is running. Later, when replaying it reads from the recording.
+
+Web Replay's strategy is similar to an end-to-end testing framework, which records user interactions and network calls and later replays the session. The major differences is the layer of abstraction (JS/OS). The test might be able to re-run consistenly, but it would be impossible to use it to step backwards in the debugger or jump back to an arbitrary point. This and many more things are now possible with Web Replay.
```
Comment (juliandescottes) [inline _posts/2018-09-18-webreplay.markdown]: The major differences is -> difference (singular) ?

File: _posts/2018-09-18-webreplay.markdown [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  return 0;
+}
+```
+
+There is some more special logic, but I think this summarized version does a good job of showing that we want to be able to call the original function, add the call to our thread's events stream, and save the output.
+
+
+### Conclusion
+
+Web Replay is super low-level, but it's architecture does not need to be unacessible. For me, it's been a great opportunity to better understand how Firefox collaborates with the OS to perform the basic functions of a browser.
```
Comment (juliandescottes) [inline _posts/2018-09-18-webreplay.markdown]: it's architecture -> its architecture?

**Review by juliandescottes — COMMENTED** [review]
reads nicely!

**Pull request merged**